bot/src/main/kotlin/me/melijn/bot/database/manager/PlaylistTrackManager.kt

```diff
@@ -0,0 +1,158 @@
+package me.melijn.bot.database.manager
+
+import kotlinx.datetime.LocalDateTime
+import me.melijn.ap.injector.Inject
+import me.melijn.bot.database.model.PlaylistTrack
+import me.melijn.bot.model.PartialUser
+import me.melijn.bot.music.*
+import me.melijn.bot.utils.KoinUtil.inject
+import me.melijn.gen.FetchedTrackData
+import me.melijn.gen.PlaylistData
+import me.melijn.gen.PlaylistTrackData
+import me.melijn.gen.SpotifyTrackData
+import me.melijn.gen.database.manager.AbstractFetchedTrackManager
+import me.melijn.gen.database.manager.AbstractPlaylistTrackManager
+import me.melijn.gen.database.manager.AbstractSpotifyTrackManager
+import me.melijn.gen.database.manager.AbstractTrackManager
+import me.melijn.kordkommons.database.DriverManager
+import me.melijn.kordkommons.utils.TimeUtil
+import org.jetbrains.exposed.sql.*
+import java.util.*
+import me.melijn.bot.database.model.FetchedTrack as DBFetchedTrack
+import me.melijn.bot.database.model.SpotifyTrack as DBSpotifyTrack
+import me.melijn.bot.database.model.Track as DBTrack
+
+@Inject
+class PlaylistTrackManager(override val driverManager: DriverManager) : AbstractPlaylistTrackManager(driverManager) {
+
+    private val trackManager by inject<TrackManager>()
+    private val playlistFetchedTrackManager by inject<PlaylistFetchedTrackManager>()
+    private val playlistSpotifyTrackManager by inject<PlaylistSpotifyTrackManager>()
+
+    fun newTrack(playlist: PlaylistData, track: Track) {
```

Comment:
newTrack also seems to share a lot of lines with the BotRestartTrackManager file